docs/model-types.mdx

```diff
@@ -0,0 +1,467 @@
+
+Let's create and train an ML model.
+
+```sql
+CREATE MODEL mindsdb.house_sales_predictor
+FROM files
+  (SELECT * FROM house_sales)
+PREDICT MA
+ORDER BY saledate
+GROUP BY bedrooms, type
+-- the target column to be predicted stores one row per quarter
+WINDOW 8      -- using data from the last two years to make forecasts (last 8 rows)
+HORIZON 4;    -- making forecasts for the next year (next 4 rows)
+```
+
+On execution, we get:
+
+```sql
+Query successfully completed
+```
+
+Once the training process is completed, we can query for predictions.
+
+```sql
+SELECT m.saledate AS date, m.MA AS forecast, MA_explain
+FROM mindsdb.house_sales_predictor AS m
+JOIN files.house_sales AS t
+WHERE t.saledate > LATEST
+AND t.type = 'house'
+AND t.bedrooms = 2
+LIMIT 4;
+```
+
+On execution, we get:
+
+```sql
++-------------+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| date        | forecast          | MA_explain                                                                                                                                                                                    |
++-------------+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 2019-12-31  | 441413.5849598734 | {"predicted_value": 441413.5849598734, "confidence": 0.99, "anomaly": true, "truth": null, "confidence_lower_bound": 440046.28237074096, "confidence_upper_bound": 442780.88754900586}        |
+| 2020-04-01  | 443292.5194586229 | {"predicted_value": 443292.5194586229, "confidence": 0.9991, "anomaly": null, "truth": null, "confidence_lower_bound": 427609.3325864327, "confidence_upper_bound": 458975.7063308131}        |
+| 2020-07-02  | 443292.5194585953 | {"predicted_value": 443292.5194585953, "confidence": 0.9991, "anomaly": null, "truth": null, "confidence_lower_bound": 424501.59192981094, "confidence_upper_bound": 462083.4469873797}       |
+| 2020-10-02  | 443292.5194585953 | {"predicted_value": 443292.5194585953, "confidence": 0.9991, "anomaly": null, "truth": null, "confidence_lower_bound": 424501.59192981094, "confidence_upper_bound": 462083.4469873797}       |
++-------------+-------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+For details, check out [this tutorial](/sql/tutorials/house-sales-forecasting/).
+
+## How it Works in the Background
+
+MindsDB uses the Lightwood ML engine by default. The so-called mixers are used to create models by detecting patterns and behaviors of data. Here are [the mixers used by Lightwood](https://lightwood.io/mixer.html).
```

Comment:
Let's reinforce the fact this is Lightwood-exclusive and may be different in other engines:

> MindsDB uses the Lightwood ML engine by default. This section takes a closer look at how this package automatically chooses what type of model to use. 

> Models in Lightwood follow an _encoder-mixer-decoder_ pattern, where refined (i.e. encoded) representations of all features are _mixed_ to produce target predictions. Here are...

Reply:
Also, it's probably best to mention there is an ensembling step after training all mixers.

Reply:
Modified it like this:

> ## How it Works in the Background
> 
> MindsDB uses the Lightwood ML engine by default. This section takes a closer look at how this package automatically chooses what type of model to use.
> 
> Models in Lightwood follow an encoder-mixer-decoder pattern, where refined, or encoded, representations of all features are mixed to produce target predictions. Here are [the mixers used by Lightwood](https://lightwood.io/mixer.html).
> 
> Please note that there is an ensembling step after training all mixers. Read on to learn more.
> 
> To give you some details on how MindsDB creates a model using different mixers, here is [the full code](https://github.com/mindsdb/lightwood/blob/9a48d5523d78111c77024e1983d281f37a9782ea/lightwood/api/json_ai.py).